### PR TITLE
check domain_hint query param existence before adding

### PR DESCRIPTION
**Pull request by wakurth**
When adding a domain_hint to the extraQueryParameter field, the
domain_hint query parameter is appended multiple times on subsequent
requests to obtain a token in an iframe. This causes the login to fail.
This fixes checks for the existence of the domain_hint parameter prior
to adding it.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23issuecomment-130807770%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23issuecomment-130809879%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23issuecomment-130812325%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23issuecomment-133185830%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23issuecomment-134373093%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23discussion_r37579218%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23discussion_r37580351%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23discussion_r37584001%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23issuecomment-130807770%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hi%20__%40wakurth__%2C%20I%27m%20your%20friendly%20neighborhood%20Microsoft%20Pull%20Request%20Bot%20%28You%20can%20call%20me%20MSBOT%29.%20Thanks%20for%20your%20contribution%21%5Cr%5Cn%20%20%20%20%3Cspan%3E%5Cr%5Cn%20%20%20%20%20%20%20%20In%20order%20for%20us%20to%20evaluate%20and%20accept%20your%20PR%2C%20we%20ask%20that%20you%20sign%20a%20contribution%20license%20agreement.%20It%27s%20all%20electronic%20and%20will%20take%20just%20minutes.%20I%20promise%20there%27s%20no%20faxing.%20https%3A//cla.microsoft.com.%5Cr%5Cn%20%20%20%20%3C/span%3E%5Cr%5Cn%5Cr%5CnTTYL%2C%20MSBOT%3B%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-08-13T19%3A11%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9287708%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/msftclas%22%7D%7D%2C%20%7B%22body%22%3A%20%22__%40wakurth__%2C%20Thanks%20for%20signing%20the%20contribution%20license%20agreement%20so%20quickly%21%20Actual%20humans%20will%20now%20validate%20the%20agreement%20and%20then%20evaluate%20the%20PR.%5Cr%5Cn%3Cbr%20/%3EThanks%2C%20MSBOT%3B%22%2C%20%22created_at%22%3A%20%222015-08-13T19%3A20%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9287708%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/msftclas%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-08-13T19%3A28%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1013700%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wakurth%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-08-20T21%3A36%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1013700%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wakurth%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-24T20%3A39%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20ddd2738b770554352f469a0de2f9771d37651667%20lib/adal.js%20130%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/158%23discussion_r37579218%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40wakurth%20Could%20you%20comment%20this%20a%20bit.%20%20I%20don%27t%20understand%20what%20line%20331%20is%20doing.%5Cr%5Cn%5Cr%5CnWouldn%27t%20the%20regex%5Cr%5Cn%5Cr%5Cn%3E%20%5B%3F%26%5Dname%3D%5Cr%5Cn%5Cr%5CnAccomplish%20the%20same%20thing%20simpler%3F%20%20The%20capture%20group%20is%20superfluous.%20%20I%20think%20you%20can%20just%20use%20regex.test%28url%29.%22%2C%20%22created_at%22%3A%20%222015-08-20T20%3A47%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40RandalliLama%20...%20Thank%20you%20for%20the%20response.%20I%27ll%20look%20at%20simplifying%20that%2C%20and%20re-submit%20an%20update.%22%2C%20%22created_at%22%3A%20%222015-08-20T20%3A57%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1013700%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wakurth%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-08-20T21%3A35%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1013700%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/wakurth%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/adal.js%3AL327-339%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/RandalliLama%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%7D%2C%20%22https%3A//github.com/wakurth%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1013700%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/RandalliLama'></a><a href='https://github.com/wakurth'></a>

- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/158#issuecomment-130807770'>General Comment</a></b>
- <a href='https://github.com/msftclas'></a> Hi __@wakurth__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
<span>
In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
</span>
TTYL, MSBOT;
- <a href='https://github.com/msftclas'></a> __@wakurth__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;
- [x] <a href='#crh-comment-Pull ddd2738b770554352f469a0de2f9771d37651667 lib/adal.js 130'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/158#discussion_r37579218'>File: lib/adal.js:L327-339</a></b>
- <a href='https://github.com/RandalliLama'></a> @wakurth Could you comment this a bit.  I don't understand what line 331 is doing.
Wouldn't the regex
> [?&]name=
Accomplish the same thing simpler?  The capture group is superfluous.  I think you can just use regex.test(url).
- <a href='https://github.com/wakurth'></a> @RandalliLama ... Thank you for the response. I'll look at simplifying that, and re-submit an update.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/158?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/158?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/158?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/158'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/158'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by wakurth** [comment]
:shipit:

File: lib/adal.js [hunk 1]
```diff
@@ -316,6 +327,12 @@ AuthenticationContext.prototype._renewIdToken = function (callback) {
     this._loadFrame(urlNavigate, 'adalIdTokenFrame');
 };
 
+AuthenticationContext.prototype._urlContainsQueryStringParameter = function (name, url) {
+    name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+    var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
```
Comment (RandalliLama) [inline lib/adal.js]: Could you comment this a bit.  I don't understand what line 331 is doing.

Wouldn't the regex
> [?&]name=
Accomplish the same thing simpler?  The capture group is superfluous.  I think you can just use regex.test(url).

Comment (wakurth) [inline lib/adal.js]: @RandalliLama ... Thank you for the response. I'll look at simplifying that, and re-submit an update.

Comment (wakurth) [inline lib/adal.js]: :+1:

**Comment by wakurth** [comment]
:shipit:

**Comment by RandalliLama** [comment]
:shipit:

**Pull request merged**